src/VectorStore.ts

```diff
@@ -1,186 +1,292 @@
-import { TFile, Vault } from "obsidian";
+import {TFile, Vault} from 'obsidian';
+import {Md5} from 'ts-md5';
+import {
+	ChatCompletionRequestMessage,
+	ChatCompletionRequestMessageRoleEnum,
+	CreateChatCompletionResponse
+} from 'openai/api';
 // @ts-ignore
-import similarity from "compute-cosine-similarity";
-import {Throttler} from "./Throttler";
+import similarity from 'compute-cosine-similarity';
+import {CreateEmbeddingResponse} from "openai";
 
-export type Vector = Array<number>
+type Vector = Array<number>
+type Chunk = {
+	contents: string;
+	embedding: Vector;
+}
+type FileEntry = {
+	eTs: number;
+	mTs: number;
+	md5hash: string;
+	fileEmbedding: Vector;
+	chunks: Array<Chunk>;
+}
+type DatabaseFile = {
+	version: number;
+	mTs: number;
+	embeddings: Map<string, FileEntry>;
+}
+export type NearestVectorResult = {
+	path: string;
+	chunk: string | undefined;
+	similarity: number;
+}
+type FileEntryUpdate = {
+	path: string;
+	chunk: boolean;
+	hash: string | undefined;
+	contents: string;
+	embedding: Vector | undefined;
+	mTs: number | undefined;
+}
 
+const isChunk = (item: Chunk | undefined): item is Chunk => {
+	return !!item
+}
 export class VectorStore {
-	constructor(vault: Vault) {
+	private readonly dbFileName = "vault-chat.json"
```

Comment:
Renamed the db file name to vault-chat.json. This just leave the database2.json file behind which seems fine since this is beta.